en/advanced_config/ethernet_setup.md

```diff
@@ -0,0 +1,85 @@
+# PX4 Ethernet Setup
+
+Ethernet connectivity a faster, more reliable, and more flexible communication technology than serial communication approach.
+This method is preferred over serial communication in case of direct connection between Pixhawk and IP Radio telemetry, companion computers and Networks.
+It utilizes UDP connection for MAVLINK communication over the port 14550 with [MAVLINK2](https://mavlink.io/en/guide/mavlink_2.html). 
+
+## Supported Flight Controllers
+
+PX4 supports Ethernet connectivity on [Pixhawk 5X-standard](https://github.com/pixhawk/Pixhawk-Standards/blob/master/DS-011%20Pixhawk%20Autopilot%20v5X%20Standard.pdf) flight controllers that have an Ethernet port.
+It may also be supported on other boards.
+
+Supported flight controllers include:
+
+ - [Holybro Pixhawk 5X](../flight_controller/pixhawk5x.md)
+
+
+## Connecting PX4 and a Ubuntu Linux Computer
+
+To connect PX4 and a Ubuntu Linux computer you need to configure them to run on the same IP network.
+On PX4 you update this configuration in the network configuration file on the SD card (`/fs/microsd/net.cfg`).
+On Ubuntu you can do this by defining a netplan configuration file: `/etc/netplan/01-network-manager-all.yaml`
+
+Below we set up the systems on an IP network with addresses `192.168.0.Xxx`, where PX4 has address `192.168.0.4` and the computer has address `192.168.0.1`.
```

Comment:
You use the word "protocol" a lot. I think you need to be more precise - below I use the term IP address scheme -  Because what I **think** you're doing here is setting  up a local subnet with ranges `192.168.0.Xxx`. Is that right? 

However I'm a bit rusty on networks, so can you confirm that:
- `192.168.0.4` is the hard coded address we are giving the flight controller.
- If so, why are we assigning a DNS server to PX4?
- How do you decide what address range to assign - is this just based on your router?
- For the Ubuntu, I have guessed that "the computer has address `192.168.0.1`" but I don't understand netplan config files. It could be that "192.168.0.1/24" means that it can have any address from 1 to 24.
  The main point here is can you explain what that netplan does, so that perhaps we can hint to users how they configure it themselves.
  
Upshot, it looks like you set up a subnet and then manually assign IP addresses in the range for each member. No DHCP etc. Is that right?

Reply:
- yes we have made it to that custom IP with netman commands within the previous steps. However, this is hardly coded and optional which could be customized. We have just made an assumption for the users who have no idea about networking. 
- I have followed the PR for Nuttx PR for that. 
- It can be customized actually, but since users may not have experience with networking, I assumed a whole setup based on an example. 
- 24 first bits of addressing. examples are [here](https://netplan.io/examples/)

Reply:
Great thanks! What I've done here is provide links to the relevant docs and make it clear that this is an example that works with the PX4 setup. We don't need to explain everything, we just need to make it easy for people to find out.